README.md

```diff
@@ -1,19 +1,127 @@
 
-This repository has 2 different scripts that you can use with ansible.
-1. `d42_ansible_inventory_hostfile.py` can be used to create a static inventory file for ansible. You can group hosts by tags, customers, building or service level from Device42 data.
-2. `d42_ansible_dynamic_inventory.py` can be used to dynamically by ansible to get hosts from Device42 based on certain filters.
+This Ansible Collection contains the inventory and lookup plugins for Device42. 
+1. Inventory (device42.d42.d42) - can be used to dynamically generate Ansible inventory from Device42 devices.
+2. Lookup (device42.d42.d42)- This lookup plugin allows a playbook to fetch passwords or ip addresses for hosts from Device42.
+3. Lookup (device42.d42.d42_prompt) - This lookup plugin allows a playbook to fetch passwords or ip addresses for hosts from Device42 with prompt. 
 
+In the `contrib` directory you will find the legacy inventory scripts. Please favor the plugin over the legacy scripts:
+1. `contrib\inventory\d42_ansible_inventory_hostfile.py` can be used to create a static inventory file for ansible. You can group hosts by tags, customers, building or service level from Device42 data.
+2. `contrib\inventory\d42_ansible_dynamic_inventory.py` can be used to dynamically by ansible to get hosts from Device42 based on certain filters.
+
+-----------------------------
 ## Assumptions
+- This script works with Device42 12.0.0 and above
+### Requirements
+- ansible 2.9+
+- python 3.6.x+
+- requests (you can install it with pip install requests or apt-get install python-requests)
+- Ansible must have an available connection to your Device42 instance in order to collect devices for inventory
+
+### Installation
+#### Ansible Tower:
+* Clone the repo on the host of your Anisble server
+```bash
+git clone git@github.com:device42/ansible_device42.git
+```
+* Navigate to the cloned directory
+```bash
+cd path/to/directory
+```
+* Build the Ansible collection
+```bash
+ansible-galaxy collection build
+```
+* Install the Ansible collection 
+```bash
+ansible-galaxy collection install device42-d42-2.0.0.tar.gz -p ../.ansible/collections
+```
+
+* Set the following environmental variables (optional)
+```
+D42_USER = 'device42 user'
+D42_PWD = 'device42 password'
+D42_URL = 'https:// device42 server IP address'
+D42_SKIP_SSL_CHECK = False
+```
+
+For more information on installing collections please follow documentation here https://docs.ansible.com/ansible/latest/user_guide/collections_using.html
+
 -----------------------------
-    * This script works with Device42 12.0.0 and above
+## Inventory Plugin
+
+### Configuration
+Define an inventory file (`*.d42.yml`) within .ansible/collections/ansible_collections/device42/d42/plugins/inventory or wherever your
```

Comment:
The inventory file (`*.d42.yml`) can be located anywhere. Ansible will lookup the plugin specified on the `plugin: ` line of the inventory file from the collection if it is installed in the COLLECTION_PATH

The lookups for the collection will first look relative to the playbook in `collections/ansible_collections/`. Then it will try `~/.ansible/collections/ansible_collections` then `/usr/share/ansible/collections/ansible_collections`. 

For example: if you are running `ansible-playbook -i d42.yml my-playbook.yml` it will first look to `./collections/ansible_collections/`. If you are testing with the `ansible-inventory` command you will need to provide the `--playbook-dir` option to look relative to the inventory file. `ansible-inventory -i d42.yml --list --playbook-dir=./`